src/errors.rs

```diff
@@ -2,24 +2,57 @@ use error_chain::error_chain;
 
 error_chain! {
     foreign_links {
-        Clap(::clap::Error);
-        Io(::std::io::Error);
-        SyntectError(::syntect::LoadingError);
-        ParseIntError(::std::num::ParseIntError);
-        GlobParsingError(::globset::Error);
+        Clap(clap::Error);
+        Io(std::io::Error);
+        SyntectError(syntect::LoadingError);
+        ParseIntError(std::num::ParseIntError);
+        GlobParsingError(globset::Error);
     }
 }
 
 pub fn default_error_handler(error: &Error) {
     match error {
         Error(ErrorKind::Io(ref io_error), _)
-            if io_error.kind() == ::std::io::ErrorKind::BrokenPipe =>
+            if io_error.kind() == std::io::ErrorKind::BrokenPipe =>
         {
-            ::std::process::exit(0);
+            std::process::exit(0);
         }
         _ => {
             use ansi_term::Colour::Red;
             eprintln!("{}: {}", Red.paint("[bat error]"), error);
         }
     };
 }
+
+// Mock out a type for clap::Error if we aren't pulling in a dependency on clap.
```

Comment:
So I guess it's not possible to put a feature gate on the `Clap(clap::Error);` entry in `foreign_links`? Would it be possible to put one on the full `error_chain!` macro call (which would result in some duplication, but maybe still better than a fake type)?

Reply:
Ah I tracked down the right syntax, they just do it backwards with the cfg after the variant instead of in front. :thinking: 

```rust
foreign_links {
    Clap(::clap::Error) #[cfg(feature = "application")];
}
```